torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -526,3 +526,302 @@ def forward(self, *inputs: Any) -> Any:
+
+        if not image_or_videos:
+            raise TypeError("Found no image in the sample.")
+        if len(image_or_videos) > 1:
+            raise TypeError(
+                f"Auto augment transformations are only properly defined for a single image or video, "
+                f"but found {len(image_or_videos)}."
+            )
+        if not bboxes_list:
+            raise TypeError("Found no bounding box in the sample.")
+        if len(bboxes_list) > 1:
+            raise TypeError(
+                f"Auto augment transformations are only properly defined for a single bboxes tensor, "
+                f"but found {len(bboxes_list)}."
+            )
+
+        idx1, image_or_video = image_or_videos[0]
+        idx2, bboxes = bboxes_list[0]
+        return (flat_inputs, spec, idx1, idx2), image_or_video, bboxes
+
+    def _unflatten_and_insert_image_or_video_and_bboxes(
+        self,
+        flat_inputs_with_spec: Tuple[List[Any], TreeSpec, int, int],
+        image_or_video: Union[datapoints.ImageType, datapoints.VideoType],
+        bboxes: datapoints.BoundingBox,
+    ) -> Any:
+        flat_inputs, spec, idx1, idx2 = flat_inputs_with_spec
+        flat_inputs[idx1] = image_or_video
+        flat_inputs[idx2] = bboxes
+        return tree_unflatten(flat_inputs, spec)
+
+    @staticmethod
+    def _transform_image_or_video_in_bboxes(
+        fn: Callable[..., torch.Tensor], fn_kwrgs: dict, image: torch.Tensor, bboxes: torch.Tensor
+    ) -> torch.Tensor:
+        new_img = image.clone()
+        for bbox in bboxes:
+            bbox_img = new_img[..., bbox[1] : bbox[3], bbox[0] : bbox[2]]
+            out_bbox_img = fn(bbox_img, **fn_kwrgs)
+            new_img[..., bbox[1] : bbox[3], bbox[0] : bbox[2]] = out_bbox_img
+        return new_img
+
+    def _apply_image_or_video_and_bboxes_transform(
+        self,
+        image: Any,
+        bboxes: datapoints.BoundingBox,
+        transform_id: str,
+        magnitude: float,
+        interpolation: InterpolationMode,
+        fill: Dict[Type, datapoints.FillTypeJIT],
+    ) -> Tuple[Any, datapoints.BoundingBox]:
+        # TODO: SolarizeAdd, Cutout, BBox_Cutout, Flip_Only_BBoxes, Cutout_Only_BBoxes
+
+        if transform_id.endswith("_Only_BBoxes"):
+            transform_id = transform_id[:-12]
+            fn_kwargs = dict(transform_id=transform_id, magnitude=magnitude, interpolation=interpolation, fill=fill)
+            image = self._transform_image_or_video_in_bboxes(
+                self._apply_image_or_video_transform, fn_kwargs, image, bboxes
+            )
+        elif transform_id.endswith("_BBox"):
+            transform_id = transform_id[:-5]
+            image = self._apply_image_or_video_transform(image, transform_id, magnitude, interpolation, fill)
+
+            if transform_id == "Rotate":
+                bboxes = bboxes.rotate(angle=magnitude)
```

Comment:
Actually, I'm not quite sure to understand why should we call transforms separately on bboxes. Can't we group them together with image ?

Reply:
To confirm, "group them together with image" means modifying method `_AutoAugmentBase._apply_image_or_video_transform` by adding an optional `bbox` argument. Is that correct? (I previously avoid touching that class.)